CHANGES.md

```diff
@@ -1,3 +1,11 @@
+0.8.0
+=====
+
+- Global variables contained in pickle strings will override existing
+  variables when loaded in their new environment. This restores the (previously
+  untested) behavior of cloudpickle prior to changes done in 0.6 and 0.6.1.
```

Comment:
Actually by re-reading the changelog, I think the original unwanted change was probably introduced in 0.5.4.

This is confirmed by the following script that used to pass in 0.5.3 and breaks after 0.5.4 and is passes again with the current PR.

```python
from loky import get_reusable_executor

e = get_reusable_executor()

def f1(x):
    return -x

def f2(x):
    return f1(x)

assert e.submit(f2, 1).result() == -1

def f1(x):
    return x

assert e.submit(f2, 1).result() == 1
```

I will add this to the loky test suite. We don't have any test in cloudpickle directly that tests the repeated execution of a dynamically defined functions with code that changes over time in a long running children process.

I think this is missing, because this is the main use case for cloudpickle but this requires new test infrastructures (similar to our `subprocess_pickle_echo` helper).

Reply:
Yes, the change happened in `0.5.4` for functions defined in `__main__`, and in `0.6.0` for the ones defined in other modules.
Regarding your test, I agree that having a`subprocess_pickle_echo` tied to a specific python process would be nice to have.

Reply:
What you are testing is close to `test_interactively_defined_global_variable` and `test_closure_interacting_with_a_global_variable` (testing if a global variable is properly overridden), but are more realistic use-cases from a end-user point of view. Were you thinking about refactoring these tests, or adding a new one?

Reply:
But rather than echoing an arbitrary pickled object, I would rather call a callable pickled with its argument and return a pickle of the result of the call which is different from calling the result of a callable passed to `subprocess_pickle_echo`.